### DISCUSSION TITLE
Status Update & Project Goals

### DISCUSSION THREAD

**Discussion by binyamin** [binyamin/eleventy-garden#71] [Meta]
I've been thinking that a template should focus on a single goal. The goal of this project is to aid in building a mind-garden. Changes to the project should support that goal.

I'd like to make a list of 2 (or 3) features which are important for a mind-garden to have, and stick to those. Please leave suggestions below.

**My List**
1. Backlinks
2. Wikilinks
3. Tags

**My anti-list**
1. Search
2. Blog
3. Static pages
4. Works with Obsidian, Roam, etc.
5. Page Preview

**Comment by binyamin** [comment]
Thoughts welcome @asbjornb @tannerdolby @gerwitz @RosieCampbell

**Reply by asbjornb** [reply]
I'm sorry I didn't reply to this. I don't have much input on which direction this project should take. I think it is already fine for a simple plug&play mind-garden. A bit more info might be needed but I haven't tried hosting yet or anything so I'm not sure - and would be happy to help build the guides if I find something lacking.

Personally I want notes as a part of a personal website. So for me adding pages and maybe something akin to blogposts will be neccesary. I don't think that has to be included here though - as long as we can link to some info on building it out. Backlinks, wikilinks, tags all sound like reasonable features. A note-graph showing connections (like in foam or others) would be great too but I haven't much idea how to implement it.

So again - if just a plug&play garden is wanted this is a great starting place for anyone wanting to use 11ty. However I think that most people after starting here would want to build a bit more on their site and if that is the case it would be great if there was a way to build out but still rely on updates to the core functionality from this (or another) repo. Like you mentioned in #79 that might be possible by making this a plugin rather than a template but any options would be great.

**Reply by binyamin** [reply]
@asbjornb Not sure if I told you about this yet. I recently built a plugin for backlinks (See #84), like you suggested. https://github.com/binyamin/eleventy-plugin-backlinks

**Comment by xplosionmind** [comment]
I have a very complex and heavily customized Eleventy website, and I am one of those who definitely cannot start from this template and build everything else from the bottom-up.

In light of this, I believe more comprehensive information and tutorials should be available, for people who intend to manually implement wikilinking and backlinking functionality to their websites.

For example, a few questions I have in mind:
- does the backlinking implementation work only with wikilinks, or it does with any kind of url?
- does this template use Nunjucks or Liquid by default?

I would be really grateful not only if you could answer to these questions, but provide more context to the implementation of the digital garden features that this website takes advantage of.

Thanks a lot!

Best,  
Tommi

**Reply by asbjornb** [reply]
binyamin knows this much better than I and I'm sure he can provide a more complete answer. 

I'm sure you could use this functionality in a larger site. There are a few elements to it though and I'm no expert, but having this template at hand take a look at:
.eleventy.js: line 3-29 sets up markdownit (check their github) with options to recognize and redirect wiki linkes (line 14-20) and to use it for converting markdown files in 11ty. Also on line 31-33 it makes sure to add all notes to the 'notes' collection
./layouts/note.html adds the backlink section at ./includes/backlinks.html - which shows incoming links to note pages below the content.
./notes/notes.11tydata.js is the heavy lifter. It ensures all files in that directory uses the note.html layout, but then it also goes on to construct a preview of the content if anything links to it, and it adds itself to any linking notes' backlinks.

Most of the rest in the template is just a basic setup to showcase it - though you'll definitely need the markdownit dependency from package.json too if going this route.

Does this help?

(There are also more complete mindgarden utils, templates and frameworks available for gatsby and some other ssg's but if you are already on 11ty or have a preference for it I think this should be a good start for some simple wiki-links functionality).

**Reply by xplosionmind** [reply]
> Does this help?

Yes, it does! Thanks a lot, @asbjornb.

In any case, I openes #84 exactly to better tackle the issues I noted.
